roomserver/internal/query/query.go

```diff
@@ -237,7 +237,11 @@ func (r *Queryer) QueryMembershipAtEvent(
 	}
 
 	for _, eventID := range request.EventIDs {
-		stateEntry := stateEntries[eventID]
+		stateEntry, ok := stateEntries[eventID]
+		if !ok {
+			response.Memberships[eventID] = []*gomatrixserverlib.HeaderedEvent{}
```

Comment:
Please update the docs for `QueryMembershipAtEvent` to say that sending back `[]` is possible.